Detectors/TPC/reconstruction/macro/convertTracks.C

```diff
@@ -0,0 +1,123 @@
+
+using namespace o2::TPC;
+using namespace o2::Base::Track;
+
+struct OutputTrack
+{
+	float Alpha;
+	float X;
+	float Y;
+	float Z;
+	float SinPhi;
+	float DzDs;
+	float QPt;
+	int NClusters;
+	int FitOK;
+};
+
+struct EventHeader
+{
+  int run;
+
+};
+
+void convertTracks(TString inputBinaryFile, TString inputClusters, TString outputFile)
+{
+
+  // ===| input chain initialisation |==========================================
+  TChain c("cbmsim");
+  c.AddFile(inputClusters);
+
+  TClonesArray *clusters=0x0;
+  c.SetBranchAddress("TPCCluster", &clusters);
+  //c.SetBranchAddress("TPC_Cluster", &clusters);
+
+  // ===| output tree |=========================================================
+  TFile fout(outputFile, "recreate");
+  TTree tout("events","events");
+
+  // ---| output data |---------------------------------------------------------
+  vector<TrackTPC> arrTracks;
+  //TClonesArray *arrTracksPtr = new TClonesArray("TrackTPC");
+  //TClonesArray &arrTracks = *arrTracksPtr;
+  EventHeader eventHeader;
+
+  tout.Branch("header", &eventHeader, "run/I");
+  tout.Branch("Tracks", &arrTracks);
+  
+  // ===| input binary file |===================================================
+  FILE* fpInput = fopen(inputBinaryFile, "rb");
+  if (fpInput == NULL)
+  {
+    printf("Error opening input file\n");
+    exit(1);
+  }
+
+  // ===| Loop over all events in the input file |==============================
+  //Number of events is not stored int the file, but we just read until we reach the end of file.
+  //All values stored in the file are either int or float, so 32 bit.
+  //Thus, we do not need to care about alignment. The same data structures must be used again for reading the file.
+  int nEvents = 0;
+  std::vector<unsigned int> ClusterIDs;
+  while (!feof(fpInput))
+  {
+    int numTracks; //Must be int!
+    OutputTrack track;
+    size_t count = fread(&numTracks, sizeof(numTracks), 1, fpInput);
+    if (!count) break;
+    printf("Event: %d, Number of tracks: %d, %zu\n", nEvents, numTracks, count);
+
+    // ---| set event information |---------------------------------------------
+    eventHeader.run = 12345;
+
+    // ---| read cluster tree |-------------------------------------------------
+    c.GetEntry(nEvents);
+
+    // ---| loop over tracks |--------------------------------------------------
+    for (int iTrack = 0;iTrack < numTracks;iTrack++)
+    {
+      arrTracks.clear();
+      count = fread(&track, sizeof(track), 1, fpInput);
+      printf("Track %d Parameters: Alpha %f, X %f, Y %f, Z %f, SinPhi %f, DzDs %f, Q/Pt %f, Number of clusters %d, Fit OK %d\n", iTrack, track.Alpha, track.X, track.Y, track.Z, track.SinPhi, track.DzDs, track.QPt, track.NClusters, track.FitOK);
+
+      //TrackTPC* track = new(arrTracks[iTrack]) TrackTPC();
+      TrackTPC trackTPC(track.X, track.Alpha, {track.Y, track.Z, track.SinPhi, track.DzDs, track.QPt});
+      arrTracks.push_back(trackTPC);
+
```

Comment:
@wiechula : here the push_back will add a copy of the the trackTPC to the vector.
Later you add clusters to trackTPC and not its copy which was added to arrTracks.
The TrackTPC class has no copy c-tor, so in the shallow copy of trackTPC the TrackTPC::mClusterArray  will point to the TClonesArray of the local trackTPC you have in the loop. In the end of the loop this TClonesArray will be deleted (see following comment on TrackTPC class) which will invalidate also the cluster array of the track added to arrTracks.
To avoid this, it is better to do all following operation (cluster addition) on the copy of track added to the arrTracks: 
     TrackTPC & storedTrack = arrTracks.back();
      ...
      storedTrack.AddCluster(...)